llpc/test/shaderdb/OpLoad_StackBoundsCheck.pipe

```diff
@@ -0,0 +1,123 @@
+; BEGIN_SHADERTEST
+; RUN: amdllpc -spvgen-dir=%spvgendir% -v %gfxip %s | FileCheck -check-prefix=SHADERTEST %s
+; SHADERTEST-LABEL: {{^// LLPC}} pipeline patching results
+
+; SHADERTEST: = icmp ult i32 %[[#]], 16
+; SHADERTEST: br i1 %
+
+; SHADERTEST: = getelementptr [16 x <2 x float>], [16 x <2 x float>] addrspace(5)* %
+; SHADERTEST: = load float, float addrspace(5)* %
+; SHADERTEST: br label %
+
+; SHADERTEST: = phi float [ %[[#]], %[[#]] ], [ 0.000000e+00, %.entry ]
+
+; SHADERTEST-LABEL: {{^// LLPC}} final pipeline module info
+; SHADERTEST: AMDLLPC SUCCESS
+; END_SHADERTEST
+
+[Version]
+version = 40
+
+[VsGlsl]
+#version 450 core
+
+layout(location = 0) in vec4 inV0;
+
+layout(location = 0) out vec4 out1;
+
+void main(void)
+{
+    gl_Position = inV0;
+    out1 = inV0;
+}
+
+[VsInfo]
+entryPoint = main
+
+[FsGlsl]
+#version 450 core
+
+layout(location = 0) in vec4 color_in;
+layout(location = 0) out vec4 color_out;
+
+void main() {
+  vec2 localData[16];
+  for (int i = 0; i < 16; i++)
+    localData[i] = vec2(color_in.x + i + 5);
+  // Read out-of-bounds of the stack array
+  float v = localData[int(color_in.x * 255) + 5024].x;
```

Comment:
Just based on this test, it's unclear to me if this will insert one check in the loop (times unroll count) and one outside, or just outside. Can you add some comments that relates the checks to the code, or perhaps put the checks just above the code they correspond to?

Reply:
For the loop, llvm can statically show that all accesses are in-bounds so it will remove the checks. The only check left is the one that is actually out-of-bounds.
I think most (all?) shaderdb tests have their checks at the top or bottom of the file, should I move them around anyway?

Reply:
> I think most (all?) shaderdb tests have their checks at the top or bottom of the file, should I move them around anyway?
LIT doesn't care. Is the other (shadertest?) tool still in use? Not sure what limitations it has.

Reply:
IMO, It would be really valuable to have LGC tests for this, where you can see the IR before and after the pass.